src/NodaTime.Test/DateIntervalTest.cs

```diff
@@ -169,6 +171,423 @@ public void Equals_DifferentToOtherType()
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_Extremes_Unequal_DifferentCalendars()
+        {
+            LocalDate start1 = new LocalDate(2000, 1, 1);
+            LocalDate end1 = new LocalDate(2001, 6, 19);
+            LocalDate start2 = start1.WithCalendar(CalendarSystem.Gregorian);
+            LocalDate end2 = end1.WithCalendar(CalendarSystem.Gregorian);
+            var interval1 = new DateInterval(start1, end1, true);
+            var interval2 = new DateInterval(start2, end2, true);
+            Assert.AreNotEqual(interval1, interval2);
+
+            // Neither interval above can be represented as an exclusive interval, but they must still compare unequal.
+            IEqualityComparer<DateInterval> comparer = DateInterval.NormalizingEqualityComparer;
+            Assert.IsFalse(comparer.Equals(interval1, interval2));
+            Assert.AreNotEqual(comparer.GetHashCode(interval1), comparer.GetHashCode(interval2));  // probably
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_EmptyRange()
+        {
+            LocalDate date1 = new LocalDate(2000, 1, 1);
+            LocalDate date2 = new LocalDate(2000, 1, 2);
+            var interval1 = new DateInterval(date1, date1, false);
+            var interval2 = new DateInterval(date2, date2, false);
+            Assert.AreEqual(0, interval2.Length);
+            Assert.AreEqual(0, interval1.Length);
+            Assert.AreNotEqual(interval1, interval2);
+
+            IEqualityComparer<DateInterval> comparer = DateInterval.NormalizingEqualityComparer;
+            Assert.IsFalse(comparer.Equals(interval1, interval2));
+            Assert.AreNotEqual(comparer.GetHashCode(interval1), comparer.GetHashCode(interval2));  // probably
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_NullToNonNull()
+        {
+            LocalDate start = new LocalDate(2000, 1, 1);
+            LocalDate end = new LocalDate(2001, 6, 19);
+            var interval = new DateInterval(start, end, false);
+            Assert.IsFalse(DateInterval.NormalizingEqualityComparer.Equals(interval, null));
+            Assert.IsFalse(DateInterval.NormalizingEqualityComparer.Equals(null, interval));
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_NullToNull()
+        {
+            Assert.IsTrue(DateInterval.NormalizingEqualityComparer.Equals(null, null));
+        }
+
+        [Test]
+        public void NormalizingEqualityComparer_DateIntervalToItself()
+        {
+            LocalDate start = new LocalDate(2000, 1, 1);
+            LocalDate end = new LocalDate(2001, 6, 19);
+            var interval = new DateInterval(start, end, false);
+            Assert.IsTrue(DateInterval.NormalizingEqualityComparer.Equals(interval, interval));
+        }
+
+        [Test]
+        public void ContainedDatesEqualityComparer_EqualValues()
```

Comment:
Are these the same tests as above, just with the different comparer? If so, I wonder whether it might be clearer to have a single set of tests, each of which check all comparers.

I suspect that with a bit of work, using parameterized test cases might be simpler too, but I'm not as sure about that.